nav2_dwb_controller/dwb_critics/test/obstacle_footprint_test.cpp

```diff
@@ -0,0 +1,255 @@
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <vector>
+#include <memory>
+#include <string>
+
+#include "gtest/gtest.h"
+#include "rclcpp/rclcpp.hpp"
+#include "dwb_critics/obstacle_footprint.hpp"
+#include "dwb_core/exceptions.hpp"
+
+class OpenObstacleFootprintCritic : public dwb_critics::ObstacleFootprintCritic
+{
+public:
+  double pointCost(int x, int y)
+  {
+    return dwb_critics::ObstacleFootprintCritic::pointCost(x, y);
+  }
+
+  double lineCost(int x0, int x1, int y0, int y1)
+  {
+    return dwb_critics::ObstacleFootprintCritic::lineCost(x0, x1, y0, y1);
+  }
+};
+
+geometry_msgs::msg::Point rotate_origin(geometry_msgs::msg::Point p, double angle)
+{
+  double s = sin(angle);
+  double c = cos(angle);
+
+  // rotate point
+  double xnew = p.x * c - p.y * s;
+  double ynew = p.x * s + p.y * c;
+
+  p.x = xnew;
+  p.y = ynew;
+
+  return p;
+}
+
+geometry_msgs::msg::Point p(double x, double y)
+{
+  geometry_msgs::msg::Point p;
+  p.x = x;
+  p.y = y;
+  return p;
+}
+
+// Variables
+double footprint_size_x_half = 1.8;
+double footprint_size_y_half = 1.6;
+
+std::vector<geometry_msgs::msg::Point> getFootprint()
+{
+  std::vector<geometry_msgs::msg::Point> footprint;
+  footprint.push_back(p(footprint_size_x_half, footprint_size_y_half));
+  footprint.push_back(p(footprint_size_x_half, -footprint_size_y_half));
+  footprint.push_back(p(-footprint_size_x_half, -footprint_size_y_half));
+  footprint.push_back(p(-footprint_size_x_half, footprint_size_y_half));
+  return footprint;
+}
+
+TEST(ObstacleFootprint, GetOrientedFootprint)
+{
+  double theta = 0.1234;
+
+  std::vector<geometry_msgs::msg::Point> footprint_before = getFootprint();
+
+  std::vector<geometry_msgs::msg::Point> footprint_after;
+  geometry_msgs::msg::Pose2D pose;
+  pose.theta = theta;
+  footprint_after = dwb_critics::getOrientedFootprint(pose, footprint_before);
+
+  uint i;
+  for (i = 0; i < footprint_before.size(); i++) {
+    ASSERT_EQ(rotate_origin(footprint_before[i], theta), footprint_after[i]);
+  }
+
+  theta = 5.123;
+  pose.theta = theta;
+  footprint_after = dwb_critics::getOrientedFootprint(pose, footprint_before);
+
+  for (unsigned int i = 0; i < footprint_before.size(); i++) {
+    ASSERT_EQ(rotate_origin(footprint_before[i], theta), footprint_after[i]);
+  }
+}
+
+TEST(ObstacleFootprint, Prepare)
+{
+  std::shared_ptr<dwb_critics::ObstacleFootprintCritic> critic =
+    std::make_shared<dwb_critics::ObstacleFootprintCritic>();
+
+  auto node = nav2_util::LifecycleNode::make_shared("costmap_tester");
+
+  auto costmap_ros = std::make_shared<nav2_costmap_2d::Costmap2DROS>("test_global_costmap");
+  // costmap_ros->activate();
```

Comment:
remove commented out line, just to make sure someone doesn't do something later like uncomment it thinking its useful.